Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -137,6 +141,36 @@ The following command is listing shared libraries:
 $ otool -L <binary>
 ```
 
+#### Retrieving Strings
+
+Strings are always a good starting point while analyzing a binary, as it provides a context to the associated code. For instance, an error log string "Cryptogram generation failed." gives us a hint that the adjoining code might be responsible for generation of a cryptogram.
+
+Strings can be extracted from an iOS binary using a GUI tools like Ghidra or Cutter, or using a CLI-based tool such as *strings* Unix utility (`strings <path_to_binary>`) or radare2's rabin2 (`rabin2 -zz <path_to_binary>`). When using the CLI-based tools, one can use it in conjunction with other tools like *grep* or *sort* to further filter and analyze the results.
+
+#### Cross References
+
+Ghidra can be used for analyzing the iOS binaries and obtain cross references by right clicking the desired function and selecting **Show References to**.
+
+#### API Usage
+
+The iOS platform provides many in-built libraries for frequently used functionalities in applications, for example cryptography, Bluetooth, NFC, network or location libraries. Determining the presence of these libraries in an application can give us valuable information about its underlying working.
+
+For instance, if an application is importing `CC_SHA256` function, it indicates the application will be performing some kind of hashing operation using SHA256 algorithm. Further information on how to analyze iOS's cryptographic APIs is discussed in the section "[iOS Cryptographic APIs](0x06e-testing-cryptography "iOS Cryptographic APIs")".
+
+Similarly, the above approach can be used to determine where and how an application is using Bluetooth. For instance, an application performing communication using Bluetooth channel must use functions from Core Bluetooth framework like `CBCentralManager` or `connect`. Using the [iOS Bluetooth documentation](https://developer.apple.com/documentation/corebluetooth "iOS Bluetooth documenation") one can determine the critical functions and start analysis around such function imports.
+
+#### Network Communication
+
+Most of the apps you might encounter connect to remote endpoints. Even before you perform any dynamic analysis (e.g. traffic capture and analysis), you can obtain some initial inputs or entry points by enumerating the domains to which the application is supposed to communicate to.
+
+Typically these domains will be present as strings within the binary of the application. One can extract domains by retrieving strings (as discussed above) or checking the strings using tools like Ghidra. The latter option has a clear advantage: it can provide you with context, as you'll be able to see in which context each domain is being used by checking the cross-references.
```

Comment:
This one's a bit weird. The section above explains that you can get strings with both (strings command and gui stuff), and the cross-reference section is on the same level as 'retrieving strings', so not really related to it. Maybe we should remove the cross-reference titel (and thus link it to the strings part) and then this advantage can also be put there.

Reply:
Currently its there to keep iOS and Android structure same, something pointed out to me in past. Given there is only a single line of content under "cross-references" heading, I dont mind removing it. @cpholguera what is your opinion on this one?

Reply:
The Cross References section should remain since it's an important part and it's also on the Android chapter. In this chapter it should highlight what's different/special in comparison to Android (in contrast). How's following xrefs in Swift vs objc vs native? maybe a bit rephrasing would help. The point here is to tell how you find an interesting string and use "data xrefs" to get where you want, see in which context is used, from there you'd keep using "code xrefs" to get to other related functions. I guess that's what you wanted to transmit here and in the xrefs section. Was it?

Reply:
Can you use Ghidra to plot a xref data graph? that would be nice to show this point, showing where a string is being used, in this case an URL string. I know you can do it with radare2, but I'm not sure with Ghidra.

Reply:
I kept this sub-section aligned with android. And since the section is about "basic information gathering" I kept it simple. Ghidra is used in other places in the chapter after this section. 

The example of using xref to determine where the strings is used, is demonstrated in "Reviewing Disassembled Objective-C and Swift Code" section.  

I can expand this section more, but i feel the information might be repetitive in nature.